### DISCUSSION TITLE
Job Auto-Deletion

### DISCUSSION THREAD

**Discussion by SweetSoccerMoves** [jhuckaby/Cronicle#678] [Q&A] [answered]
Hello Good Sirs and 'Madams,
Preface: I'm not quite sure if this question falls under the scope of the intended use case of Cronicle, so it's not quite a feature request.
Question: Is there a way to auto-delete a job after it has run on a specific date and time?
Example: If I create a "shell script" job to run at 2023, November, 14, 5:05pm, I would like the job to run then be removed from the job list. I'm ok with an answer "Add a line in the script to delete the job"; I would just need to know where to look in the database. Thanks in advance.
-SweetSoccerMoves

**Comment by jhuckaby** [comment, marked as answer]
The only way to do it in Cronicle v1 is to script the API: https://github.com/jhuckaby/Cronicle/blob/master/docs/APIReference.md#delete_event

You can do this in a shell script using `curl`.

Cronicle v2 has a "Self Destruct" option you can enable on an event, which does exactly what you want, however Cronicle v2 is not going to be released until early 2024.

**Reply by SweetSoccerMoves** [reply]
Awesome. Thank you for the quick and detailed answer.

I saw the "delete_event" section in the APIReference guide and played around with it. What made me ask is that the EventID is only generated after saving the job; which would require me to edit the event, doubling my efforts. Thanks again and I look forward to v2. Keep up the awesome work, sir.
